lib/service/cfg.go

```diff
@@ -276,6 +276,28 @@ func (cfg *Config) ApplyToken(token string) (bool, error) {
 	return false, nil
 }
 
+// ApplyCAPins assigns the given CA pin(s), but only if each pin is not empty.
+func (cfg *Config) ApplyCAPins(caPins []string) error {
```

Comment:
I think we can remove the second loop and the `if caPins[i] != ""` because `filepath.IsAbs(value)` should handle that corner case
What do you think?

Reply:
TryReadFileAsValue will just return the string if it doesn't look like an absolute path. I think we should ignore blank lines in case a user has something like this in their config:

```
  ca_pin:
    - /Users/gavin/foo
    -
```

We were already doing the same thing for auth_token. I could condense it to one loop though, like this:

```go
	filteredPins := make([]string, 0, len(caPins))
	for _, pin := range caPins {
		if pin != "" {
			var err error
			pin, err = utils.TryReadValueAsFile(pin)
			if err != nil {
				return trace.Wrap(err)
			}
			filteredPins = append(filteredPins, pin)
		}
	}
```

Reply:
Looks good :+1: